.github/workflows/arm64.yml

```diff
@@ -37,10 +37,16 @@ jobs:
       with:
         go-version: '1.22'
         check-latest: true
+    - name: Get buf version
+      shell: bash
+      run: |
+        echo BUF_VERSION=$(go mod edit -json | jq -r ".Require.[] | select(.Path == \"github.com/bufbuild/buf\") | .Version") >> $GITHUB_ENV
     - name: Install buf cli
-      uses: bufbuild/buf-setup-action@v1
-      with:
-        github_token: ${{ github.token }}
+      run: |
+        go install github.com/bufbuild/buf/cmd/buf@${BUF_VERSION}
```

Comment:
It seems like a lot of effort 🤷 . Could keep it simple and `@main` and then `go get`. 

But I get what you;re trying to do.

Reply:
Ah, it's targeting a branch.

Reply:
I wanted to get this in place so going forward we'd only have a single version to track if we need to upgrade to a newer dev version. I don't like pulling in `main` as it isn't a reproducible build.

Reply:
Gotcha, disregard my initial comment. This is actually a good idea 😄